src/common/sync_manager.py

```
@@ -0,0 +1,632 @@
+            image.save(buf, format="PNG", optimize=True)
+            data = buf.getvalue()
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.settimeout(5.0)
+            sock.connect((self._peer_ip, self.port + 1))
+            sock.sendall(len(data).to_bytes(4, "big") + data)
+            sock.close()
+            self.logger.info(
+                "Sync: sent scroll image %dx%d (%d bytes compressed)",
+                image.width, image.height, len(data),
+            )
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)